src/commands/configuration-management/node.service.ts

```diff
@@ -0,0 +1,41 @@
+import { NodeApi } from "./api/node-api";
+import { Context } from "../../core/command/cli-context";
+import { fileService, FileService } from "../../core/utils/file-service";
+import { logger } from "../../core/utils/logger";
+import { v4 as uuidv4 } from "uuid";
+
+export class NodeService {
+    private nodeApi: NodeApi;
+
+    constructor(context: Context) {
+        this.nodeApi = new NodeApi(context);
+    }
+
+    public async findNode(packageKey: string, nodeKey: string, withConfiguration: boolean, jsonResponse: boolean): Promise<void> {
+        const node = await this.nodeApi.findStagingNodeByKey(packageKey, nodeKey, withConfiguration);
+
+        if (jsonResponse) {
+            const filename = uuidv4() + ".json";
+            fileService.writeToFileWithGivenName(JSON.stringify(node, null, 2), filename);
+            logger.info(FileService.fileDownloadedMessage + filename);
+        } else {
+            logger.info(`ID: ${node.id}`);
+            logger.info(`Key: ${node.key}`);
+            logger.info(`Name: ${node.name}`);
+            logger.info(`Type: ${node.type}`);
+            logger.info(`Package Node Key: ${node.packageNodeKey}`);
+            if (node.parentNodeKey) {
+                logger.info(`Parent Node Key: ${node.parentNodeKey}`);
+            }
+            logger.info(`Created By: ${node.createdBy}`);
+            logger.info(`Updated By: ${node.updatedBy}`);
+            logger.info(`Creation Date: ${new Date(node.creationDate).toISOString()}`);
+            logger.info(`Change Date: ${new Date(node.changeDate).toISOString()}`);
+            if (node.configuration) {
```

Comment:
What if the node has invalid configuration? Should we add an if condition that checks whether the node is invalid and then show the configuration from the appropriate field based on that?